src/components/Component/Component.tsx

```diff
@@ -10,20 +10,20 @@ import {
 import createComponent, { CreateComponentReturnType } from '../../lib/createComponent'
 import { ReactProps } from '../../../types/utils'
 
-export interface SlotProps
-  extends UIComponentProps<SlotProps>,
+export interface ComponentProps
+  extends UIComponentProps<ComponentProps>,
```

Comment:
unfortunately, with the changes introduced this type becomes to be very hard to read

Reply:
Yes, Component vs UIComponent is not so clear.  How about if UIComponent becomes BaseComponent?

Another option would be to call `Component` `Box` instead... ?

Reply:
agreed, see https://github.com/stardust-ui/react/pull/713/#discussion_r247552617

Reply:
@levithomason, would rather choose second option (`Component` -> `Box`), as this provide more specific semantics to this component (instead of `Component` name that, essentially, doesn't  provide any). Thanks!